test/lib/usd/utils/CMakeLists.txt

```diff
@@ -33,3 +33,62 @@ mayaUsd_add_test(${TARGET_NAME}
     ENV
         "LD_LIBRARY_PATH=${ADDITIONAL_LD_LIBRARY_PATH}"
 )
+
+set(TARGET_NAME MayaTransformAPI)
+
+add_executable(${TARGET_NAME})
+
+# -----------------------------------------------------------------------------
+# sources
+# -----------------------------------------------------------------------------
+target_sources(${TARGET_NAME}
+    PRIVATE
+        main.cpp
+        test_MayaTransformAPI.cpp
+)
+add_definitions(
+  -DUSDUTILS_TEST_DATA="${CMAKE_CURRENT_SOURCE_DIR}/test_data"
+)
+
+mayaUsd_copyDirectory(${TARGET_NAME} DESTINATION ${CMAKE_CURRENT_BINARY_DIR}
+                             DIRECTORY test_data)
+
+# -----------------------------------------------------------------------------
+# compiler configuration
+# -----------------------------------------------------------------------------
+mayaUsd_compile_config(${TARGET_NAME})
+
+# -----------------------------------------------------------------------------
+# link libraries
+# -----------------------------------------------------------------------------
+target_link_libraries(${TARGET_NAME}
+    PRIVATE 
+        GTest::GTest
+        mayaUsdUtils
+        tf
+        sdf
+        ${Boost_PYTHON_LIBRARY}
+        ${python_LIBRARIES}
+)
+
+# -----------------------------------------------------------------------------
+# unit tests
+# -----------------------------------------------------------------------------
+mayaUsd_add_test(${TARGET_NAME}
+    COMMAND $<TARGET_FILE:${TARGET_NAME}>
+)
+
+# -----------------------------------------------------------------------------
+# python tests - using older approach as mayaUsd_add_test wants to use mayapy
+# -----------------------------------------------------------------------------
+
+set(USDUTILS_PYTHON_TEST_NAME Python:MayaTransformAPI)
+add_test(
+  NAME ${USDUTILS_PYTHON_TEST_NAME}
+  COMMAND
+    python -m unittest discover -s ${CMAKE_CURRENT_SOURCE_DIR}
+)
+set_tests_properties(${USDUTILS_PYTHON_TEST_NAME}
+  PROPERTIES
+    ENVIRONMENT PYTHONPATH=${CMAKE_BINARY_DIR}/${INSTALL_DIR_SUFFIX}/usdutils:${CMAKE_BINARY_DIR}/${INSTALL_DIR_SUFFIX}/lib/usd/utils:$ENV{PYTHONPATH} 
+)
```

Comment:
We shouldn't use `add_test` directly here. I see your comment `using older approach as mayaUsd_add_test wants to use mayapy`. What is the problem with mayapy?

Nit-pick: I would also remove the comment lines 81-83

Reply:
Hi @HamedSabri-adsk - As this library doesn't require maya, we didn't want to have a dependency on maya in order to be able to run the tests. However this work was done quite a while ago, and it's now apparent that mayaUsd_add_test can be passed a COMMAND argument which doesn't require maya or mayapy, we can update

Reply:
Thanks @murphyeoin for clarifying. Makes sense to me now.